docs/GUIDE.md

```diff
@@ -0,0 +1,175 @@
+# Welcome to the OVAL-visualization-as-graph guide
+
+This tool generates a tree graph from ARF xml report from OpenSCAP scan.
+
+## Prerequisites
+
+### Minimal requirements
+
+- **python3**
+- [lxml](https://pypi.org/project/lxml/)
+
+### Recommended requirements
+
+- includes minimal requirements
+- [inquirer](https://pypi.org/project/inquirer/)
+
+## Installation
+
+### Fedora 30 and later
+
+```bash
+sudo dnf install oval-graph
+# Install inquirer, if you want have nice cli features. (optional)
+sudo pip3 install inquirer
+```
+
+### Installation via [pip](https://pypi.org/project/oval-graph/)
+
+```bash
+# If you want to install oval-graph to $HOME/.local/bin, you have to run the below command:
+pip3 install --user oval-graph
+# Install inquirer, if you want have nice cli features. (optional)
+pip3 install --user inquirer
+
+# If you want to install oval-graph globally instead, you have to run the below commands as admin, e.g. on Linux:
+sudo pip3 install oval-graph
+# Install inquirer, if you want have nice cli features. (optional)
+sudo pip3 install inquirer
+
+```
+
+### Installation from source
+
+```bash
+git clone https://github.com/OpenSCAP/OVAL-visualization-as-graph.git
+cd OVAL-visualization-as-graph
+
+# Install without features (light version)
+sudo pip3 install .
+
+# OR
+
+# Install with nice cli features
+sudo pip3 install ".[niceCli]"
+```
+
+> If you don't want to install anything, look [here](#run-from-source).
+
+## Commands
+
+- [arf-to-graph](#arf-to-graph) Generates a tree graph from ARF xml report
+- [arf-to-json](#arf-to-json) Generates a JSON from ARF xml report
+- [json-to-graph](#json-to-graph) Generates a tree graph from JSON
+
+### Positional arguments
+
+- `source_filename` ARF scan a file or a JSON file
+- `rule_id` Rule ID to be process. A part from the full rule ID, a part of the ID or a regular expression can be used. If brackets are used in the regular expression the regular expression must be quoted.
+
+### Optional parameter
+
+- `-h, --help` Shows help message and exit
+- `--all` Processes all matched rules
+- `--hide-passing-tests` Do not display passing tests for better orientation in graphs that contain a large amount of nodes. [example](#option-hide-passing-tests) (not implemented in `arf-to-json`)
+- `-o OUTPUT, --output OUTPUT` The directory where to save output directory with files.
+- `--off-web-browser` It does not start the web browser. Default opens the web browser with graph report. Default is Firefox. If Firefox is not installed it opens default web browser in OS. (not implemented in `arf-to-json`)
+
+#### Optional parameters to make it easier to find a visualized rule
+
+These parameters are useful when the user does not know the full name of the rule or wants to reduce the selection. (not implemented in `json-to-graph`)
+
+- `--show-fail-rules` Show only FAIL rules
```

Comment:
Unrelated, but it should be `--show-failed-rules` :-)

Reply:
I create issue for this. #96